### PR TITLE
Add more information to data request result

### LINKED ISSUE #625: Add more information to data request result (chain side)
We want to have more data the result of data requests, including:
- Timestamp of data request
- Timestamp of data aggregation
- Number of requested validators
- Number of sufficient validators
- Number of reported validators

All tests must be updated too.

### PR DESCRIPTION
**Pull request by prin-r**
Fix #625

### CODE REVIEW

File: chain/x/zoracle/internal/keeper/result_test.go [hunk 1]
```diff
@@ -24,12 +33,27 @@ func TestAddResultSuccess(t *testing.T) {
 	_, err := keeper.GetResult(ctx, 1, 1, []byte("calldata"))
 	require.NotNil(t, err)
 
+	keeper.SetRequest(ctx, 1, types.NewRequest(
```
Comment (taobun) [inline chain/x/zoracle/internal/keeper/result_test.go:37]: Can you set BlockTime to context, It makes `AggregationTime` test look more solid.

File: chain/x/zoracle/handler_test.go [hunk 2]
```diff
@@ -369,7 +369,17 @@ func TestEndBlock(t *testing.T) {
 
 	result, err := keeper.GetResult(ctx, 1, 1, calldata)
 	require.Nil(t, err)
-	require.Equal(t, []byte("answer2"), result)
+	require.Equal(t,
```
Comment (taobun) [inline chain/x/zoracle/handler_test.go:373]: Set BlockTime before handleEndBlock?

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/result.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package types
+
+// Result is a data structure that stores the detail of a result of a specific request.
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/result.go]: of the result of a specific request.

**Review by sorawit — CHANGES_REQUESTED** [review]
Are there changes required on the client side (how to get proof?)

**Review by taobun — DISMISSED** [review]

**Comment by taobun** [comment]
Implement `evmBytesProof` will be in the next pr?

File: chain/x/zoracle/internal/types/result.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		AggregationTime:          aggregationTime,
+		RequestedValidatorsCount: requestedValidatorsCount,
+		SufficientValidatorCount: sufficientValidatorCount,
+		ReportedValidatorsCount:  reportedValidatorsCount,
+		Data:                     data,
+	}
+}
+
+// DecodeResult is a helper function for decoding bytes to Result.
+func DecodeResult(b []byte) Result {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/result.go]: This function should return (Result, error) or should panic if the condition hits.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/result.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		RequestedValidatorsCount: requestedValidatorsCount,
+		SufficientValidatorCount: sufficientValidatorCount,
+		ReportedValidatorsCount:  reportedValidatorsCount,
+		Data:                     data,
+	}
+}
+
+// DecodeResult is a helper function for decoding bytes to Result.
+func DecodeResult(b []byte) (Result, error) {
+	if len(b) <= 40 {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/result.go]: <

File: chain/x/zoracle/internal/types/result.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		return Result{}, fmt.Errorf(fmt.Sprintf("expect size of input to be more than 40 bytes but got %d bytes", len(b)))
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/result.go]: Expect .... to be at least ...

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**